Chatto/Source/ChatController/BaseChatViewController+Scrolling.swift

```diff
@@ -113,6 +113,38 @@ extension BaseChatViewController {
         collectionView.contentOffset = CGPoint(x: 0, y: collectionView.contentOffset.y + diffY)
     }
 
+    public func scrollToItem(withId itemId: String, position: UICollectionView.ScrollPosition = .centeredVertically, animated: Bool = false) {
+        guard let collectionView = self.collectionView else { return }
+        guard let itemIndex = self.chatItemCompanionCollection.indexOf(itemId) else { return }
+
+        let indexPath = IndexPath(item: itemIndex, section: 0)
+        guard let rect = self.rectAtIndexPath(indexPath) else { return }
+
+        if animated {
+            let pageSize = collectionView.bounds.height
+            let isScrollingUp = rect.minY < collectionView.contentOffset.y
+
+            if isScrollingUp {
+                let isNeedToScrollUpMoreThenTwoPages = rect.minY < collectionView.contentOffset.y - 2 * pageSize
```

Comment:
"- 2" is a bit annoying and unclear. let's extract it as a variable with a readable name?